dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,323 @@
+import math
+import numpy as np
+import toolz
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+from sklearn.utils import check_random_state
+
+from tornado import gen
+from toolz import groupby
+from dask.distributed import as_completed, default_client, futures_of
+from dask_searchcv.model_selection import DaskBaseSearchCV
+import dask.array as da
+from ..datasets import make_classification
+from ..wrappers import Incremental
+from ._split import train_test_split
+from time import time
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_meta_future, X, y, meta=None, fit_params={},
+                 dry_run=False, **kwargs):
+    start = time()
+    assert isinstance(X, np.ndarray)
+    assert y is None or isinstance(y, np.ndarray)
+    model, m = model_meta_future
+    if meta is None:
+        meta = m
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(X, y, **fit_params)
```

Comment:
What is `dry_run`?  Can it be removed?

Reply:
There is, and it's already part of the testing framework. This is a good reorganization.